src/Native/Unix/System.Native/pal_process.c

```diff
@@ -339,7 +339,22 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
     // on another thread while a vfork() child is still pending, bad things are possible; however we
     // do not do that.
 
+#if defined (__GLIBC__)
     if ((processId = vfork()) == 0) // processId == 0 if this is child process
+#else
+    // musl libc has an undocumented failure mode around setuid(); we must exclude it.
+    // Don't pack this expression down. setjmp rules apply. See K&R C 2.0 Appendix B.8
```

Comment:
I'm not following this second sentence.  Can you elaborate?

Reply:
The C specification is weak here. I'm not certain `pid = ((condition) ? vfork() : (fork());` is safe. This is the reason I didn't write the test at all the first time.

Reply:
More to the point, I spent hours verifying the `if/else` was safe and I didn't have to use a volatile function pointer (which is trivial to prove safe but very ugly). I'm not willing to do so again for ?: to save three lines of code.

Reply:
> I'm not willing to do so again for ?: to save three lines of code.

No one's along you to :) But I think you can delete the comment.